### PR TITLE
Env multip poll

### PR DESCRIPTION
**Pull request by hzhuang1**

### CODE REVIEW

File: wd_util.c [hunk 1]
```diff
@@ -1037,12 +1077,17 @@ static int wd_init_async_polling_thread_per_numa(struct wd_env_config *config,
...
 	struct async_task_queue *task_queue, *head;
-	int i, j, ret;
+	int i, j, n, ret;
 
 	if (!config_numa->async_ctx_num)
 		return 0;
 
-	config_numa->async_poll_num = WD_ASYNC_DEF_POLL_NUM;
+	if (config_numa->async_poll_num <= 0) {
+		printf("Invalid async poll number (%ld) is set.\n",
```
Comment (lin755) [inline wd_util.c:1086]: WD_ERR is better

File: wd_util.c [hunk 2]
```diff
@@ -1037,12 +1077,17 @@ static int wd_init_async_polling_thread_per_numa(struct wd_env_config *config,
...
+			config_numa->async_poll_num);
+		printf("Change to default value: %d\n", WD_ASYNC_DEF_POLL_NUM);
```
Comment (lin755) [inline wd_util.c:1088]: same

File: wd_util.c [hunk 3]
```diff
@@ -1051,7 +1096,11 @@ static int wd_init_async_polling_thread_per_numa(struct wd_env_config *config,
 	head = task_queue;
 	config_numa->async_task_queue_array = (void *)head;
 
-	for (i = 0; i < config_numa->async_poll_num; task_queue++, i++) {
+	if (config_numa->async_poll_num > config_numa->async_ctx_num)
+		n = config_numa->async_ctx_num;
+	else
+		n = config_numa->async_poll_num;
```
Comment (lin755) [inline wd_util.c:1102]: Use  a  max function from standard c library better here?

**Comment by hzhuang1** [comment]
@wangzhou @haofang111 @lin755 @youngersh @Liulongfang Could you help to share any comments?

**Review by lin755 — APPROVED** [review]
LGTM

File: wd_util.c [hunk 4]
```diff
@@ -1051,7 +1096,13 @@ static int wd_init_async_polling_thread_per_numa(struct wd_env_config *config,
 	head = task_queue;
 	config_numa->async_task_queue_array = (void *)head;
 
-	for (i = 0; i < config_numa->async_poll_num; task_queue++, i++) {
+	if (config_numa->async_poll_num > config_numa->async_ctx_num) {
+		n = config_numa->async_ctx_num;
+		WD_ERR("Create %ld async polling threads since ctx number is "
+		       "limited.\n", config_numa->async_poll_num);
```
Comment (wangzhou) [inline wd_util.c:1102]: here do you want to create poll thread which is the number of async_ctx_num when config_numa->async_poll_num > config_numa->async_ctx_num, if so, there is a wrong print

**Pull request merged**